lib/kernel/src/code.erl

```diff
@@ -488,7 +493,7 @@ clear_cache() -> call(clear_cache).
 -spec ensure_modules_loaded([Module]) ->
    'ok' | {'error',[{Module,What}]} when
       Module :: module(),
-      What :: badfile | nofile | on_load_failure.
+      What :: badfile | nofile | on_load_failure | embedded.
```

Comment:
I have updated the typespec but the embedded return was already a possibility back in OTP 25. It just was not documented.

Reply:
@bjorng since you originally implemented this function, I'd love your opinion here. There is a tiny inconsistency. This function may return `{error, embedded}` but only for modules with `on_load`. I think we should either always return `{error, embedded}` or never return `{error, embedded}`.

The simplest approach is to never return `{error, embedded}` and add an entry to the docs saying as much. What do you think?

Reply:
> The simplest approach is to never return `{error, embedded}` and add an entry to the docs saying as much. What do you think?

Yes, I think that is the best solution. There may be code that depends on `ensure_modules_loaded` working in embedded mode.

Reply:
Perfect, I pushed a fix and documented the differences.